src/diffusers/models/transformers/transformer_flux.py

```diff
@@ -125,6 +125,8 @@ def forward(
         gate = gate.unsqueeze(1)
         hidden_states = gate * self.proj_out(hidden_states)
         hidden_states = residual + hidden_states
+        if hidden_states.dtype == torch.float16:
+            hidden_states = hidden_states.clip(-65504, 65504)
```

Comment:
Could you explain why we have to do this? Can we just upcast the VAE like we do for SDXL when using FP16? 

https://github.com/huggingface/diffusers/blob/4c6152c2fb0ade468aadb417102605a07a8635d3/src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py#L1257

Reply:
its actually an overflow in the DiT, no? upcasting the VAE is useful for the VAE's attention overflows, but doesn't stop us from having a problem earlier

see SD 2.1-v and the problem with xformers that requires portion of the unet be upcast - but the vae doesn't have to be upcasted there! kinda different from SDXL's issue with activations in the VAE.

Reply:
Mine was just a question not an assertion. I didn’t know that it overflows within DiT.

Reply:
The clipped activations were just found through trial and error. I documented the statistics of each activation when running the DiT and found which ones caused fp16 overflow.
[fp16-clip-0.txt](https://github.com/user-attachments/files/16514210/fp16-clip-0.txt)
[fp16-clip-2.txt](https://github.com/user-attachments/files/16514211/fp16-clip-2.txt)
[fp16-clip-1.txt](https://github.com/user-attachments/files/16514212/fp16-clip-1.txt)

Reply:
Sorry for probably dumb question, but can't it be scaled down to fit in fp16? Because as I can see in any case this data later passed to norm? I don't know if any norm sensitive to scaling, but basic which I know sensitive only slightly due to epsilon.

Reply:
I wanted to do this but this model's norm layer runs activations through SiLU and a linear layer with bias, so no the activations can't be scaled down